### PR TITLE
Argo CD is just CD; Added Argo Workflows for CI.

### PR DESCRIPTION
**Pull request by protosam**
`Argo CI/CD` is mislabeled. Argo has multiple components. CD and CI are two separate ones. I've ensured they are both available and I've corrected the labeling.

* Argo CD - this is delivery only, not CI. As such, I've renamed `Argo CI/CD` to `Argo CD`.
* Argo Workflows - this is a jobs pipeline tool; and is the CI component for Argo. I've added this to the repo.

Tested deployment of argo-workflows:
```text
$ kubectl apply -f app.yaml 
namespace/argo created
customresourcedefinition.apiextensions.k8s.io/clusterworkflowtemplates.argoproj.io created
customresourcedefinition.apiextensions.k8s.io/cronworkflows.argoproj.io created
customresourcedefinition.apiextensions.k8s.io/workfloweventbindings.argoproj.io created
customresourcedefinition.apiextensions.k8s.io/workflows.argoproj.io created
customresourcedefinition.apiextensions.k8s.io/workflowtemplates.argoproj.io created
serviceaccount/argo created
serviceaccount/argo-server created
role.rbac.authorization.k8s.io/argo-role created
clusterrole.rbac.authorization.k8s.io/argo-aggregate-to-admin created
clusterrole.rbac.authorization.k8s.io/argo-aggregate-to-edit created
clusterrole.rbac.authorization.k8s.io/argo-aggregate-to-view created
clusterrole.rbac.authorization.k8s.io/argo-cluster-role created
clusterrole.rbac.authorization.k8s.io/argo-server-cluster-role created
rolebinding.rbac.authorization.k8s.io/argo-binding created
clusterrolebinding.rbac.authorization.k8s.io/argo-binding created
clusterrolebinding.rbac.authorization.k8s.io/argo-server-binding created
configmap/workflow-controller-configmap created
service/argo-server created
service/workflow-controller-metrics created
deployment.apps/argo-server created
deployment.apps/workflow-controller created

$ kubectl get pods -n argo
NAME                                   READY   STATUS    RESTARTS   AGE
argo-server-6cdfbbcfb5-mxzbk           0/1     Running   0          20s
workflow-controller-64c947c746-4hgm6   1/1     Running   0          20s
```

### CODE REVIEW

File: argo-workflows/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+name: argo-workflows
+title: "Argo Workflows"
+version: (default)
```
Comment (saiyam1814) [inline argo-workflows/manifest.yaml:4]: can we have a specific version instead of default ?

File: argocd/manifest.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
-name: argo
-title: "Argo CI/CD"
+name: argo-cd
+title: "Argo CD"
 version: (default)
```
Comment (saiyam1814) [inline argocd/manifest.yaml:4]: Same for this - specific version instead of default.

**Comment by protosam** [comment]
@saiyam1814, done.

**Comment by saiyam1814** [comment]
@protosam version should be quoted string like "v0.1.0"

**Comment by protosam** [comment]
@saiyam1814 done.

It functionally doesn't matter though. Semantic versioning is always going to be a string to YAML. Semantic versioning being formatted like `[0-9]+\.[0-9]+\.[0-9]+`. 


These three are just `(default)`:
```text
$ grep '^version: (default)' */manifest.yaml -l
chaos-mesh/manifest.yaml
metrics-server/manifest.yaml
traefik/manifest.yaml
```

If this is a concern though, every one of these existing manifests don't meet this requirement of being quoted for the version:
```text
$ grep '^version: [^"]' */manifest.yaml -l
argo-workflows/manifest.yaml
bitwardenrs/manifest.yaml
cert-manager/manifest.yaml
chaos-mesh/manifest.yaml
docker-registry/manifest.yaml
dynamic-pv-scaler/manifest.yaml
flux/manifest.yaml
ghost/manifest.yaml
gitea/manifest.yaml
helm/manifest.yaml
jenkins/manifest.yaml
joomla/manifest.yaml
kube-hunter/manifest.yaml
kube-scan/manifest.yaml
kubei/manifest.yaml
kubeless/manifest.yaml
kubernetes-dashboard/manifest.yaml
kubevious/manifest.yaml
linkerd/manifest.yaml
litmuschaos/manifest.yaml
loki/manifest.yaml
longhorn/manifest.yaml
maesh/manifest.yaml
mariadb/manifest.yaml
metrics-server/manifest.yaml
mongodb/manifest.yaml
netdata/manifest.yaml
nextcloud/manifest.yaml
nginx/manifest.yaml
nodered/manifest.yaml
okteto/manifest.yaml
openfaas/manifest.yaml
permission-manager/manifest.yaml
polaris/manifest.yaml
portainer/manifest.yaml
rancher/manifest.yaml
sealed-secrets/manifest.yaml
selenium/manifest.yaml
system-upgrade-controller/manifest.yaml
tekton/manifest.yaml
traefik/manifest.yaml
wordpress/manifest.yaml
```

**Comment by saiyam1814** [comment]
Yes We are slowly moving to standardize things and getting rid of default version and making sure all the new commits happen with quoted strings (I know this was fixed but just want to keep things same)
We will be reaching out to maintainers to get these fixed but for the new ones we want a specific version than default of latest. 
There is an option for plans where people can define both stable and latest in manifest and give version as multiple.

**Comment by saiyam1814** [comment]
PR lgtm

**Pull request merged**